_plugins/auto-anchor.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+## Automatically add id tags to list items that are formatted like one
+## of the following:
+# - **<Summary:** Details
+# - *<Summary:* Details
+# - [Summary][]: Details
+# - [Summary](URL): Details
+
+Jekyll::Hooks.register :documents, :pre_render do |post|
+  ## Don't process documents if YAML headers say: "auto_id: false"
+  if post.data["auto_id"] != false
+    post.content.gsub!(/^ *- .*/) do |string|
+      ## List items surrounded in bold
+      title = string.match(/\*\*.*\*\*/).to_s
+      ## List items surrounded in italics
+      title.empty? && title = string.match(/\*.*\*/).to_s
+      ## List items that are hyperlinks
+      title.empty? && title = string.match(/\[.*\][(\[]/).to_s
+
+      if title.empty?
```

Comment:
If the original `if title != ''` was correct, the equivalent would be either `if !title.empty?` or `unless title.empty?` (or if ActiveSupport is loaded, `if title.present?`)

ah Ruby, so many ways to do the same thing

Reply:
I also reversed the branches.  E.g. before it was: `if title != '' { a } else { b } end` and now it's `if title.empty? { b } else { a } end`

Reply:
My bad. Even better.